rpc/handlers.go

```diff
@@ -343,6 +379,70 @@ func (h *Handler) LegacyBlockWithTxs(id BlockID) (*BlockWithTxs, *jsonrpc.Error)
 	return block, nil
 }
 
+func AdaptReceipt(receipt *core.TransactionReceipt, txn core.Transaction,
+	finalityStatus TxnFinalityStatus, blockHash *felt.Felt, blockNumber uint64,
+) *TransactionReceipt {
+	messages := make([]*MsgToL1, len(receipt.L2ToL1Message))
+	for idx, msg := range receipt.L2ToL1Message {
+		messages[idx] = &MsgToL1{
+			To:      msg.To,
+			Payload: msg.Payload,
+			From:    msg.From,
+		}
+	}
+
+	events := make([]*Event, len(receipt.Events))
+	for idx, event := range receipt.Events {
+		events[idx] = &Event{
+			From: event.From,
+			Keys: event.Keys,
+			Data: event.Data,
+		}
+	}
+
+	var messageHash string
+	var contractAddress *felt.Felt
+	switch v := txn.(type) {
+	case *core.DeployTransaction:
+		contractAddress = v.ContractAddress
+	case *core.DeployAccountTransaction:
+		contractAddress = v.ContractAddress
+	case *core.L1HandlerTransaction:
+		messageHash = "0x" + hex.EncodeToString(v.MessageHash())
+	}
+
+	var receiptBlockNumber *uint64
```

Comment:
Q: what do you think about leaving a comment here stating pending blocks don't have block hashes or block numbers?

Reply:
Added comment